From a8a703d813df4b95824a42a6182ebbe04f0bc59a Mon Sep 17 00:00:00 2001
From: Kasalehlia <kasalehlia@clonejo.de>
Date: Sat, 10 Oct 2020 17:10:23 +0200
Subject: [PATCH 23/38] mesh-vpn: recognise parker as vpn type

This change overrides the gluon-core vpn-provider logic and makes sure
that mesh-vpn get's enabled when the user set the corresponding option
in the wizzard.
It also sets the simple-tc traffic limits.
---
 .../luasrc/lib/gluon/mesh-vpn/update-config   | 27 +++++++------------
 1 file changed, 9 insertions(+), 18 deletions(-)

diff --git a/package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config b/package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config
index 935d0b9a..a78576a4 100755
--- a/package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config
+++ b/package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config
@@ -1,7 +1,6 @@
 #!/usr/bin/lua
 
 local uci = require('simple-uci').cursor()
-local vpn_name, vpn = require('gluon.mesh-vpn').get_active_provider()
 
 local vpn_config = {
 	enabled = uci:get_bool('gluon', 'mesh_vpn', 'enabled'),
@@ -10,20 +9,12 @@ local vpn_config = {
 	limit_ingress = uci:get('gluon', 'mesh_vpn', 'limit_ingress'),
 }
 
-if vpn_name ~= 'fastd' then
-	uci:set('fastd', 'mesh_vpn', 'enabled', false)
-	uci:save('fastd')
-end
-
-if vpn_name ~= 'tunneldigger' then
-	uci:set('tunneldigger', 'mesh_vpn', 'enabled', false)
-	uci:save('tunneldigger')
-end
-
-vpn.enable(vpn_config.enabled)
-if vpn_config.limit_enabled then
-	vpn.set_limit(vpn_config.limit_ingress, vpn_config.limit_egress)
-else
-	vpn.set_limit(nil, nil)
-end
-
+uci:delete('simple-tc', 'client')
+uci:section('simple-tc', 'interface', 'client', {
+	ifname = 'br-client',
+	enabled = vpn_config.limit_enabled,
+	-- limits are flipped here
+	limit_ingress = vpn_config.limit_egress,
+	limit_egress = vpn_config.limit_ingress,
+})
+uci:commit('simple-tc')
-- 
2.34.1

